components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/MessageProcessorResource.java

```diff
@@ -0,0 +1,252 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.micro.integrator.management.apis;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.commons.json.JsonUtil;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.message.processor.MessageProcessor;
+import org.apache.synapse.message.processor.impl.failover.FailoverScheduledMessageForwardingProcessor;
+import org.apache.synapse.message.processor.impl.forwarder.ScheduledMessageForwardingProcessor;
+import org.apache.synapse.message.processor.impl.sampler.SamplingProcessor;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.inbound.endpoint.internal.http.api.APIResource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents message processor resources defined in the synapse configuration.
+ **/
+public class MessageProcessorResource extends APIResource {
+
+    private static final Log LOG = LogFactory.getLog(MessageProcessorResource.class);
+
+    //Message processor name query parameter
+    private static final String PROCESSOR_NAME_PARAM = "name";
+    //Message processor types
+    private static final String SAMPLING_PROCESSOR_TYPE = "Sampling-processor";
+    private static final String SCHEDULED_MESSAGE_FORWARDING_TYPE = "Scheduled-message-forwarding-processor";
+    private static final String FAILOVER_SCHEDULED_MESSAGE_FORWARDING_TYPE = "Failover-scheduled-message-forwarding-processor";
+    private static final String CUSTOM_PROCESSOR_TYPE = "Custom-message-processor";
+    //Message processor statuses
+    private static final String PROCESSOR_ACTIVE_STATUS = "active";
+    private static final String PROCESSOR_INACTIVE_STATUS = "inactive";
+    //Constants for message-processor JSON object
+    private static final String TYPE_PROPERTY = "type";
+    private static final String CONTAINER_PROPERTY = "artifactContainer";
+    private static final String FILE_NAME_PROPERTY = "fileName";
+    private static final String PARAMETER_PROPERTY = "parameters";
+    private static final String MESSAGE_STORE_PROPERTY = "messageStore";
+    private static final String STATUS_PROPERTY = "status";
+
+    public MessageProcessorResource(String urlTemplate) {
+        super(urlTemplate);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        Set<String> methods = new HashSet<>();
+        methods.add("GET");
+        methods.add("POST");
+        return methods;
```

Comment:
Shall we initialize only once in the constructor?

Reply:
done